### ISSUE TITLE
Actor Service Contract Not Allows Using Interfaces With Method Parameters 

### ISSUE THREAD

**Issue by jefersonmanjos** [Azure/service-fabric-services-and-actors-dotnet#100]
Hi there is way, so I can using interfaces  in my actor services contract? Something like this:

`public interface IMyActor : IActor{
		Task MyMethod(IProfile profile);
}`

Thanks

**Comment by jefersonmanjos** [comment]
@amanbha Do you know how to do this? Thanks

**Comment by amanbha** [comment]
@jefersonmanjos You can do that by using V2 remoting stack and adding CustomSerializer .
Your custom serializer will just add your concrete types to known types for DataContract serializer.
You can find a sample here:  https://github.com/amanbha/servicefabric/tree/master/samples/ActorCustomSerialization

**Comment by jefersonmanjos** [comment]
@amanbha thanks man. Now I will look for a way to integrate this solution with autofac. We use autofac to register ours actors.

**Comment by amanbha** [comment]
@jefersonmanjos  If you pass a wrapper type to your method that would also work without the need of custom serializer.

`
    public interface IMyType
    {
    }

   [DataContract]
    class MyType : IMyType
    {
    }
    [DataContract]
    [KnownType(typeof(MyType))]
    public class Wrapper
    {
        [DataMember]
        public IMyType Value { get; set; }
    }
`
Then yous method can accept Wrapper as an argument instead of IMyType

**Comment by jefersonmanjos** [comment]
@amanbha you are right...but for each type I will need a a custom wrapper. Because I can't use generic class too. Another way is use a wrapper classe with string Value, and the serialize IMyType to json. But this is very verbose to do and also less performatic. I think the custom serializer it's a good solution if I can configure autofac.
